plugins/connection/docker.py

```diff
@@ -308,11 +330,32 @@ def put_file(self, in_path, out_path):
                 count = ' count=0'
             else:
                 count = ''
-            args = self._build_exec_cmd([self._play_context.executable, "-c", "dd of=%s bs=%s%s" % (out_path, BUFSIZE, count)])
+            if getattr(self._shell, "_IS_WINDOWS", False):
+                cmd = '''& { $strm = [Console]::OpenStandardInput(1024)
+$data = [System.Array]::CreateInstance([byte],1024)
+$fstrm = [IO.File]::OpenWrite(%s)
```

Comment:
You still need to have more logic to handle here. Right now at a quote glance I could have a path like:

```
’+ (&[ScriptBlock]::Create([[Text.Encoding]::UTF8.GetString([Convert]::FromBase64String("my code to inject")) +’
```
_Note the use of unicode smart quotes_

PowerShell (un)helpfully treats a bunch of smart quotes as the same as single quotes so even if you ran `shlex_quote` on your path it won't escape what is needed. As soon as you can escape outside of a quote you can embed whatever command you want to run here.

There are 2 choices I can see you moving to:

* Use a the same (or similar regex) to the ansible.windows [quote filter for powershell](https://github.com/ansible-collections/ansible.windows/blob/main/plugins/filter/quote.py#L63-L70) to escape the smart quotes and give you a literal value
* Base64 encode the path and decode that in your script - base64 means you no longer have to deal with quotes or unicode characters

```python
cmd = '''
...
$outPath = [Text.Encoding]::UTF8.GetString([Convert]::FromBase64String('%s'))
...
''' % (to_text(base64.b64encode(to_bytes(out_path, encoding='utf-8'))))

Both options remove the need for `shlex_quote` but I still recommend you normalize the path as you have done today.